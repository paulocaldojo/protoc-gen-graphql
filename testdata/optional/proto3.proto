syntax = "proto3";

package protoc_gen_graphql.test.optional;

option go_package = "github.com/paulocaldojo/protoc-gen-graphql/testdata/optional";

message Proto3ScalarFields {
  optional string string_field = 1;
  optional bytes bytes_field = 2;
  optional int32 int32_field = 3;
  optional uint32 uint32_field = 4;
  optional sint32 sint32_field = 5;
  optional fixed32 fixed32_field = 6;
  optional int64 int64_field = 7;
  optional uint64 uint64_field = 8;
  optional sint64 sint64_field = 9;
  optional fixed64 fixed64_field = 10;
  optional float float_field = 11;
  optional double double_field = 12;
  optional bool bool_field = 13;
  oneof oneof_field {
    string string_1 = 14;
    string string_2 = 15;
  }
  string required_string_field = 16;
}

message Proto3NestedMessage {
  Proto3SubMessage required_sub_message = 1;
  optional Proto3SubMessage optional_sub_message = 2;
}

message Proto3SubMessage {
  string required_string = 1;
  optional string optional_string = 2;
}
