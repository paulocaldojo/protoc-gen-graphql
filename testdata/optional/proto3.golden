# DO NOT EDIT! Generated by protoc-gen-graphql.

type ProtocGenGraphqlTestOptional_Proto3ScalarFields {
  stringField: String
  bytesField: String
  int32Field: Int
  uint32Field: Int
  sint32Field: Int
  fixed32Field: Int
  int64Field: Int
  uint64Field: Int
  sint64Field: Int
  fixed64Field: Int
  floatField: Float
  doubleField: Float
  boolField: Boolean
  oneofField: ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof
  requiredStringField: String!
}

"""
`ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof` represents the `oneof_field` oneof in `protoc_gen_graphql.test.optional.Proto3ScalarFields`.
"""
union ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof = ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_1 | ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_2

"""
`ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_1` represents the `string_1` oneof field in `protoc_gen_graphql.test.optional.Proto3ScalarFields`.
"""
type ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_1 {
  _typename: String
  string1: String!
}

"""
`ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_2` represents the `string_2` oneof field in `protoc_gen_graphql.test.optional.Proto3ScalarFields`.
"""
type ProtocGenGraphqlTestOptional_Proto3ScalarFields_OneofFieldOneof_String_2 {
  _typename: String
  string2: String!
}

type ProtocGenGraphqlTestOptional_Proto3NestedMessage {
  requiredSubMessage: ProtocGenGraphqlTestOptional_Proto3SubMessage!
  optionalSubMessage: ProtocGenGraphqlTestOptional_Proto3SubMessage
}

type ProtocGenGraphqlTestOptional_Proto3SubMessage {
  requiredString: String!
  optionalString: String
}
